sdk/storage/azure-storage-common-data-movement/src/test/java/com/azure/storage/common/data/movement/PathScannerTest.groovy

```diff
@@ -0,0 +1,95 @@
+package com.azure.storage.common.data.movement
+
+import reactor.core.publisher.Flux
+import reactor.test.StepVerifier
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.attribute.AclEntry
+import java.nio.file.attribute.AclEntryPermission
+import java.nio.file.attribute.AclEntryType
+import java.nio.file.attribute.AclFileAttributeView
+import java.nio.file.attribute.PosixFilePermission
+import java.nio.file.attribute.PosixFilePermissions
+import java.nio.file.attribute.UserPrincipal
+
+class PathScannerTest extends Specification {
+    def setup() {
+        String fullTestName = specificationContext.getCurrentIteration().getName().replace(' ', '').toLowerCase()
+        String className = specificationContext.getCurrentSpec().getName()
+        // Print out the test name to create breadcrumbs in our test logging in case anything hangs.
+        System.out.printf("========================= %s.%s =========================%n", className, fullTestName)
+    }
```

Comment:
@gapra-msft we should spend some time investigating if this can be centralized. This isn't DRY and kind of clutters the codebase.
I was playing with `Global Extensions` (see https://spockframework.org/spock/docs/1.1/extensions.html) but ran into some classpath problems in commons... might be worth revisiting this.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/22448